scripts/dependencies/relock_dependencies.sh

```diff
@@ -1,17 +1,33 @@
 #!/usr/bin/env bash
 
-# can change output file names with rebuild_pipenv.sh <prefix>
+# Parse optional flag -k, to be used when we want to base the process on an existing Pipfile.lock
+KEEP_LOCK=false
+OPTIND=1
+while getopts 'k' opt; do
+    case $opt in
+        k) KEEP_LOCK=true ;;
```

Comment:
@KPrasch , @cygnusv the `Pipfile.lock` does not contain the the dependencies for the docs-requirements file - is this an issue if it it kept and reused? There is currently no `Pipfile.lock` for the docs requirements kept in the repos, only a `Pipfile`. So running the script and using an old `Pipfile.lock` would keep the dev/standard requirement files the same, but the `docs-requirements.txt` would still be freshly generated...?

Reply:
Depends on how "proper" we want to be about Pipenv.  In practice the `Pipfile.lock` is only used for `pipenv install` (and other pipenv management commands).  Since we're typically using the `Pipfile` to generate a fresh lock, and ultimately a requirements file, I'm indifferent here.

> is this an issue if it it kept and reused?

No, but it's not much of a help either.

Reply:
For regular relocking, yes no problem there. But if this script is used for a nightly build, and the dev/standard requirements are kept the same ('k' option used), and the docs-requirements are updated to newer version, does that lead to any pip installation issues, like what we observed when the docs-requirements were not relocked at the same time as the dev/standard requirements i.e. issues with `pip install -e .[dev]` prior to PR #2510 ? Or is the nightly build sufficiently different in some way that this problem won't rear its ugly head?

Reply:
@derekpierre running the script with the old pipfile is mainly motivated as a way to validate that the requirement files are consistent with the pipfile. The docs requirements are not validated, so it's not much of a problem.

Reply:
👍